packages/web-extension/src/messaging/remoteApi.ts

```diff
@@ -92,12 +106,66 @@ const consumeMethod =
     return result;
   };
 
+interface ConsumeFactoryProps<T> {
+  method: string;
+  apiProperties: RemoteApiProperties<T>;
+  getErrorPrototype: GetErrorPrototype | undefined;
+}
+
+let factoryChannelNo = 0;
+const consumeFactory =
+  <T>(
+    { method, apiProperties, getErrorPrototype }: ConsumeFactoryProps<T>,
+    { logger, messenger, destructor }: ConsumeMessengerApiDependencies
+  ) =>
+  (...args: unknown[]) => {
+    const channel = `${method}-${factoryChannelNo++}`;
+    const postSubscription = messenger
+      .postMessage({
+        factoryCall: { args: toSerializableObject(args), channel, method },
+        messageId: newMessageId()
+      } as FactoryCallMessage)
+      .subscribe();
+    const apiMessenger = messenger.deriveChannel(channel);
+    // eslint-disable-next-line no-use-before-define
+    const api = consumeMessengerRemoteApi(
+      {
+        getErrorPrototype,
+        properties: apiProperties
+      },
+      {
+        destructor,
+        logger,
+        messenger: apiMessenger
+      }
+    );
+    destructor.onGarbageCollected(api, factoryChannelNo, () => {
+      if (apiMessenger.isShutdown) {
+        return;
+      }
+      apiMessenger
+        .postMessage({
+          messageId: newMessageId(),
+          subscribe: false
+        } as CompletionMessage)
+        .subscribe(() => {
+          postSubscription.unsubscribe();
+          apiMessenger.shutdown();
+        });
+    });
+
+    // Since it returns synchronously, we can't catch potential errors.
+    // If factory method throws on the remote side, it might be a good idea to "brick" the returned api object:
```

Comment:
What’s the alternative? Impact of not implementing? Sounds fine to do this though

Reply:
This is the impact: https://github.com/input-output-hk/cardano-js-sdk/pull/723/files#diff-5250162cad8f88072beedc14e94ffdf4325136c0acb62e48dfaee349484b34b5R103

We can implement the bricking later if we find we need it